indexer/src/transform.test.ts

```diff
@@ -0,0 +1,606 @@
+        ],
+        index: 0,
+      },
+      {
+        fromAddress:
+          "0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7",
+        keys: [
+          "0x134692b230b9e1ffa39098904722134159652b09c5bc41d88d6698779d228ff",
+        ],
+        data: [
+          "0x1daabdc9a68ac6094ae282891913ac2470424809556564445182a4207f77355",
+          "0x11c5faab8a76b3caff6e243b8d13059a7fb723a0ca12bbaadde95fb9e501bda",
+          "0xffffffffffffffffffffffffffffffff",
+          "0xffffffffffffffffffffffffffffffff",
+        ],
+        index: 1,
+      },
+      {
+        fromAddress:
+          "0x4f57d7d15274d9fe832c88aafd80ecc90a8db5696913ddc39cf24af4eba3538",
+        keys: [
+          "0x5ad857f66a5b55f1301ff1ed7e098ac6d4433148f0b72ebc4a2945ab85ad53",
+        ],
+        data: [
+          "0x20",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x0",
+          "0x7e",
+          "0x1",
+          "0x13842",
+        ],
+        index: 0,
+      },
+    ],
+  };
+  const events: EventWithTransaction[] = [
+    {
+      transaction,
+      receipt: receiptWithoutLog,
+      event,
+    },
+  ];
+  const transactions: TransactionWithReceipt[] = [
+    {
+      transaction,
+      receipt: receiptWithoutLog,
+    },
+  ];
+
+  const { result, expectedHeader, ethTx, ethLogs, ethReceipt } =
+    await transformData(
+      header,
+      events,
+      transactions,
+      transaction,
+      receiptWithoutLog,
+      event,
+    );
+
+  assertEquals(result.length, 3);
+  if ("tx" in result[0].data) assertEquals(result[0].data.tx, ethTx);
+  else throw new Error("Expected tx not found in result data");
+  if ("receipt" in result[1].data) {
+    assertEquals(result[1].data.receipt, ethReceipt);
+  } else throw new Error("Expected receipt not found in result data");
+  if ("header" in result[2].data) {
+    assertEquals(result[2].data.header.number, expectedHeader.number);
+    assertEquals(result[2].data.header.hash, expectedHeader.hash);
+    assertEquals(result[2].data.header.parentHash, expectedHeader.parentHash);
+    assertEquals(result[2].data.header.sha3Uncles, expectedHeader.sha3Uncles);
+    assertEquals(result[2].data.header.stateRoot, expectedHeader.stateRoot);
+    assertEquals(result[2].data.header.gasLimit, expectedHeader.gasLimit);
+    assertEquals(
+      result[2].data.header.withdrawalsRoot,
+      expectedHeader.withdrawalsRoot,
+    );
+    assertEquals(result[2].data.header.transactions.length, 0);
```

Comment:
Here same, can't we assert on the whole object or is it too complicate? 

My idea is just to minimize the number of assert lines and possibly avoid missing a field that we could not miss if we make an assertion on the object in its entirety.

Reply:
same here:
I didn't compare the entire object because I wanted to avoid re-implementing all the logic needed to generate the Trie and Bloom data correctly on the test. I mocked those parts instead. So, I only asserted the relevant data for the header that wasn't 0x. what do you think?